src/node_jsvmapi.cc

```diff
@@ -514,6 +514,10 @@ namespace v8impl {
 #define CHECK_NEW_FROM_UTF8(isolate, result, str)                       \
     CHECK_NEW_FROM_UTF8_LEN((isolate), (result), (str), -1)
 
+#define GET_RETURN_STATUS()                                             \
+  v8impl::TryCatch::lastException().IsEmpty()                           \
```

Comment:
Good point. Problem with this: tryCatch in NAPI_PREAMBLE is defined inside the do {} while(0). Sorry, didn't catch this when refactoring that macro. I propose removing the do {} while(0) which un-breaks the tryCatch local. Don't think it makes sense to provide a separate macro or put only some of the lines of NAPI_PREAMBLE inside a scope.